src/sentry/services/hybrid_cloud/user/service.py

```diff
@@ -133,15 +138,26 @@ def get_user_by_social_auth(
     def get_first_superuser(self) -> RpcUser | None:
         pass
 
-    @rpc_method
-    @abstractmethod
     def get_or_create_user_by_email(
         self,
         *,
         email: str,
         ident: str | None = None,
         referrer: str | None = None,
     ) -> tuple[RpcUser, bool]:
+        """DEPRECATED. To be removed after usage in getsentry is replaced."""
```

Comment:
So yeah, I dislike this but AFAIK it's the only way to swap it out in `getsentry`. The flow would be

1. Deploy this PR on `sentry`, adding the "v2" method without changing the contract of the original ("v1").
2. Make a PR on `getsentry` that calls v2 instead of v1, unpacking the return value appropriately. Deploy it.
3. Only after the change from 2 is fully rolled out, make a PR on `sentry` that changes the v1 method to return `RpcUserCreationResult`. Mark v2 as a deprecated alias the delegates to v1. Deploy it.
4. Only after the change from 3 is fully rolled out, make a PR on `getsentry` calls the original method name instead of the one with `_v2` at the end. Deploy it.
5. Only after the change from 4 is fully rolled out, make a PR on `sentry` that deletes the deprecated v2 alias.

We could do this in fewer steps if, instead of "`get_or_create_user_by_email_v2`", the new method had an alternative name that we keep permanently. I contemplated "`get_or_create_user_with_email`". But considering that the "by email" phrasing is used elsewhere, I suppose it's worth doing the little deployment dance to keep the name we actually want.

Let me know if you can think of a more convenient way to do this.

Reply:
> So yeah, I dislike this but AFAIK it's the only way to swap it out in getsentry. The flow would be

I think you have the steps mostly correct. Between 1 and 2 you also need a deploy to ensure that the method is available when the getsentry change deploys.